src/neuroglancer/ui/user_events.ts

```diff
@@ -0,0 +1,220 @@
+import {RefCounted} from 'neuroglancer/util/disposable';
+
+interface TypedEventListener<T> extends EventListener {
+  (e: T): void;
```

Comment:
I'm thinking about allowing additional context to be generated here. Instead of returning the raw event (MouseEvent, WheelEvent, etc), it would return something like this:

```
interface NeuromancerEvent<EventType, ContextType> {
  rawEvent: EventType;
  context: ContextType;
}

interface TypedEventListener<E, C> extends EventListener {
 (e: NeuromancerEvent<E, C>): void;
}

type ListenerConfig<E, C> = TypedEventListener<E, C>|[TypedEventListener<E, C>, boolean];
```

Which then could be used like so:

```
interface MouseContext {
  layerId: string;
  active: boolean; // <-- e.g. from the mouse 
  // Other things that would be useful, e.g. fields from Viewer.mouseState
}

export interface DomEvents {
  mousemove?: ListenerConfig<MouseEvent, MouseContext>;
}
```